core/net/mac/contikimac/contikimac-async.c

```diff
@@ -0,0 +1,1438 @@
+static int call_off(void (*cb)(enum cm_event, rtimer_clock_t));
+static void off_done_cb(struct radio_async_cb_info info, void *ptr);
+
+/* task scheduling and locking functionality */
+static int tx_is_done(void);
+static int tx_waiting(void);
+static void release_driver_lock(void);
+static int acquire_driver_lock(int owner);
+static void schedule_next_cycle(void);
+
+/* tx related functionality */
+static PT_THREAD(strobe_sm(enum cm_event ev, rtimer_clock_t ev_time));
+static int init_send(struct mac_info info, bool receiver_awake);
+static void finish_strobes(void);
+static void start_strobes(void);
+static int tx_strobe(void);
+static void tx_done_cb(struct radio_async_cb_info info, void *ptr);
+static void set_tx_state(enum tx_state state);
+static int wait_ack(rtimer_clock_t time);
+static int inter_strobe_wait(rtimer_clock_t time);
+static void rtimer_strobe_sm(struct rtimer *t, void *ptr);
+static void cmd_cb_strobe_sm(enum cm_event ev, rtimer_clock_t ev_time);
+static void continue_ack_wait(enum cm_event ev, rtimer_clock_t ev_time);
+static void ack_done_cb(uint8_t seqno);
+static int strobes_expired(rtimer_clock_t time);
+static void strobe_cca_cb(enum cm_event ev, rtimer_clock_t ev_time);
+static int strobe_cca(void);
+static int tx_keep_going(void);
+static int strobe_cca_sleep(rtimer_clock_t now);
+static int strobe_cca_wait_stable(rtimer_clock_t now);
+
+/* rx related functionality */
+static PT_THREAD(cca_sm(enum cm_event ev, rtimer_clock_t ev_time));
+static void cmd_cb_cca_sm(enum cm_event ev, rtimer_clock_t ev_time);
+static void rtimer_call_cca_sm(struct rtimer *t, void *ptr);
+static int cca_sleep(rtimer_clock_t now);
+static int cca_wait_stable(rtimer_clock_t now);
+static void cycle_start(struct rtimer *t, void *ptr);
+static int cca_wait_check(rtimer_clock_t now);
+static bool cca_time_expired(rtimer_clock_t time);
+static bool cca_counts_expired(int *no_pkt);
+static int schedule_rx_burst_stop(rtimer_clock_t now);
+static void rx_isr_cb(bool pending);
+static void set_packet_seen(void);
+
+/* packet manipulation */
+static int create_and_secure(struct rdc_buf_list *buf_list);
+
+/* Driver API Functions */
+static void init(void);
+static void qsend_packet(mac_callback_t sent, void *ptr);
+static void qsend_list(mac_callback_t sent, void *ptr,
+                       struct rdc_buf_list *buf_list);
+static void input_packet(void);
+static int turn_on(void);
+static int turn_off(int keep_radio_on);
+static unsigned short duty_cycle(void);
+
+/* Phase Optimization Functions */
+static void schedule_phase(void);
+static void phase_wait_done(struct rtimer *t, void *ptr);
+
+/* Driver Process */
+PROCESS(contikmac_async_process, "cm async process");
+/*---------------------------------------------------------------------------*/
+/*-------------------------------API FUNCTIONS-------------------------------*/
+/*---------------------------------------------------------------------------*/
+static void
+init(void)
```

Comment:
This is missing phase_init() which leads to the nbr_phase table being used uninitialized which in turn can confuse other users of nbr_tables. For example adding it fixes the WIP Contiki Collect with Neighbor Table problem I was seeing.